pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1072,6 +1093,82 @@ func (nc *Controller) handleDisruption(zoneToNodeConditions map[string][]*v1.Nod
 	}
 }
 
+func (nc *Controller) podUpdated(oldPod, newPod *v1.Pod) {
+	if len(newPod.Spec.NodeName) != 0 && (oldPod == nil || newPod.Spec.NodeName != oldPod.Spec.NodeName) {
+		nc.podUpdateQueue.Add(newPod)
```

Comment:
I've already wrote a long comment that putting object is not the best pattern and then realized that it actually can work here. But this is purely because a given pod can be added to the queue at most once for its whole lifetime.
So this requires an extensive comment why is it done this way.

Reply:
I've reimplemented handling the queue. Now pod key (namespace/name) is added to queue.
Passing namespace is required due to podLister specification.